### ISSUE TITLE
Modeling Ideas

### ISSUE THREAD

**Issue by kshannon** [kshannon/intracranial-hemorrhage-detection#18]
vgg16 and resnet50 as a starting point was a good idea, and it seems that other people in the field have taken similar courses of actions.
> Our proposed system for the detection and classification of ICH uses multiple ImageNet6 pretrained deep convolutional neural networks (DCNNs), a preprocessing pipeline, an atlas creation module and a prediction-basis selection module (Fig. 1). The four DCNNs used for building our model are VGG167, ResNet-508, Inception-v39 and Inception-ResNet-v210." _Lee et. al. NATURE BIOMEDICAL ENGINEERING | VOL 3 | MARCH 2019 | 173–182 |_

**Comment by kshannon** [comment]
Another good point on architecture complexity and depth, this tends to follow other research papers I have read:

> Through this study, we also gained insights into approaches for optimizing deep-learning model performance. Previous work has demonstrated that deeper neural networks deliver better visual recognition performance than shallower networks when the training dataset is kept constant29. Simply choosing the deepest network, however, was not the answer in our study. We achieved much greater performance gains by iteratively adding ICH-specific pre-processing and network optimization techniques (Fig. 3, M1 to M4) compared to the small incremental improvements obtained utilizing deeper and more complex neural networks. These results imply that application-specific customization techniques are more effective to improve performance than the choice of the underlying CNN architecture. Additionally, combining the four M4 DCNNs into an ensemble model improved performance, possibly by way of consensus agreement, in much the same way that having multiple human raters can improve accuracy.  _Lee et. al. NATURE BIOMEDICAL ENGINEERING | VOL 3 | MARCH 2019 | 173–182 |_

**Comment by kshannon** [comment]
I tried for several experiments keeping the three channels as blood, brain, soft tissue windowed, but the loss for validation and train never could get  below 0.55, I an `np.stack(img)` three times after HU transformation using slope and intercept, and then resizing using `cv2.resize()`, with `interpolation=cv2.INTER_LINEAR` vs `interpolation=cv2.INTER_AREA`. So far the loss is behaving much better, down to 0.39 1/3 through first epoch, doesn't mean much, but good start.
